Sources/AppcuesKit/Data/Models/Experience.swift

```diff
@@ -15,6 +15,19 @@ internal protocol StepModel {
     var actions: [String: [Experience.Action]] { get }
 }
 
+internal enum LossyExperience {
+    case decoded(Experience)
+    case failed(FailedExperience)
+}
+
+internal struct FailedExperience: Decodable {
+    let id: UUID
+    let name: String?
+    let type: String?
```

Comment:
Thinking ahead a bit, the easiest way to log errors is in the ExperienceRenderer (specifically `.startExperience` in the state machine). That requires an `ExperienceData` (and I don't want to change that) which in turn has an `Experience` object (and changing this in `ExperienceData` to optional or something would be a headache).

If we make `FailedExperience` here require `name: String` and `type: String` rather than have those optional, we can manually construct a skeleton Experience object out of that that would fit in `ExperienceData`. Then adding an option `error: String?` to `ExperienceData` would let the state machine check for that in `.startExperience` and then report it.

This should be pretty easy, with the only change being that we're requiring two more things in `FailedExperience` (which should be there every time and would be skipped otherwise). All the decoding errors I'm worried about (read: interested in) are within the content of the experience, not the highest level structure.

Reply:
yep, I can take a stab at it (ticket 45190)